### PR TITLE
Updated Gloo to support alltoall collective operation

### LINKED ISSUE #2148: Add Gloo support for All-to-All
#2143 adds MPI and NCCL support for all-to-all.  Gloo recently added an op for all-to-all, but our current submodule ref is behind.  We should update Gloo to the latest master, then add support for Gloo when using all-to-all.

### PR DESCRIPTION
**Pull request by tgaddair**
Fixes #2148.

### CODE REVIEW

File: test/test_mxnet.py [hunk 1]
```diff
@@ -634,10 +634,6 @@ def test_horovod_alltoall(self):
         rank = hvd.rank()
         size = hvd.size()
 
-        # This test does not apply if using gloo controller
-        if hvd.gloo_enabled():
```
Comment (romerojosh) [inline test/test_mxnet.py:638]: There are several other alltoall tests below this that still contain this skip for gloo. Please remove from the remaning alltoall tests.

File: test/test_tensorflow.py [hunk 2]
```diff
@@ -1423,10 +1423,6 @@ def test_horovod_alltoall_cpu(self):
         rank = hvd.rank()
         size = hvd.size()
 
-        # This test does not apply if using gloo controller
-        if hvd.gloo_enabled():
```
Comment (romerojosh) [inline test/test_tensorflow.py:1427]: There are several other alltoall tests below this that still contain this skip for gloo. Please remove from the remaining alltoall tests.

File: test/test_torch.py [hunk 3]
```diff
@@ -891,10 +891,6 @@ def test_horovod_alltoall(self):
         rank = hvd.rank()
         size = hvd.size()
 
-        # This test does not apply if using gloo controller
-        if hvd.gloo_enabled():
```
Comment (romerojosh) [inline test/test_torch.py:895]: There are several other alltoall tests below this that still contain this skip for gloo. Please remove from the remaining alltoall tests.

File: test/test_tensorflow.py (hunk 2 above)
Comment (tgaddair) [reply test/test_tensorflow.py:1427]: Good catch, not sure how I missed all those.  Fixed!

**Pull request merged**